AjuntamentdeBarcelona/decidim.barcelona
# Add Proposal Answers

# What and why

Added a new role `reviewer` in order to review proposals. These users should be able to mark proposals as 'Accepted' or 'Rejected'. They can also add a clarification note.

This feature has been developed as a new component `ProposalAnswerBox` which it is used both for the proposal page and for administration purposes.

Added new API Endpoints:
- `POST /proposals/:proposal_id/answers`: Create a proposal answer.
- `PATCH /proposals/:proposal_id/answers`: Update a proposal answer.
# QA

Checkout the project and run `npm install`. Then, as a `reviewer` you should be able to see the proposal answer box.
# GIF Tax

![](https://media.giphy.com/media/rcw4NNFE6JioM/giphy.gif)
# TODOs
- [x] Refactor API controllers
- [x] Create `ProposalAnswer` model
- [x] ~~Fix npm crazy stuff~~
- [x] Create API endpoints
- [x] Create `ProposalAnswerBox` component
- [x] Add new role (`reviewer`) and verify permissions
- [x] Styles magic
- [x] Loading
- [x] Reenable buttons
- [x] i18n XD
